examples/simple_goto/simple_goto.py

```diff
@@ -27,14 +27,16 @@ def arm_and_takeoff(aTargetAltitude):
     while vehicle.gps_0.fix_type < 2:
         print "Waiting for GPS...:", vehicle.gps_0.fix_type
         time.sleep(1)
-		
+        
     print "Arming motors"
     # Copter should arm in GUIDED mode
     vehicle.mode    = VehicleMode("GUIDED")
     vehicle.armed   = True
     vehicle.flush()
 
-    while not vehicle.armed and not api.exit:
+    while not vehicle.armed:
+        if api.exit: #check to make sure mavproxy hasn't commanded us to stop
+            api.stop()
```

Comment:
@djnugent This changes is fine by me. When we have a callback on api.exit things will be much cleaner of course since you won't have to have this checker code everywhere if you're always going to be killing the thread.